geci_distance/geci_distance.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import pandas as pd
+import numpy as np
+from lmfit import Model
+from scipy import integrate
+
+
+def hazard_model(x, sigma, beta):
+    return 1 - np.exp(-((x / sigma) ** (-beta)))
+
+
+class GECI_Distance:
+    def __init__(self, n_obs, distances, n_bins=10):
+        self.distances = distances
+        self.n_bins = n_bins
+        self.n_obs = n_obs
+
+    def set_line_width(self, width):
+        self.width = width
+
+    def set_line_length(self, length):
+        self.length = length
+
+    def set_study_area(self, area, units="km2"):
+        if units == "km2":
+            self.area = area * 1_000_000
+        if units == "m2":
+            self.area = area
+
+    def initialize_hazard_model(self):
+        model = Model(hazard_model)
+        model.set_param_hint("sigma", value=1, min=0)
+        model.set_param_hint("beta", value=1, min=0)
+        params = model.make_params()
+        return model, params
+
+    def calculate_mid_points(self, bins):
+        return (bins[:-1] + bins[1:]) / 2
+
+    def normalize_histogram(self, pdf):
+        return pdf / pdf[0]
+
+    def fit_detection_function(self, model="hazard"):
+        if model == "hazard":
+            model, params = self.initialize_hazard_model()
+            result = model.fit(self.norm_hist, params, x=self.bins_mid_points, method="nelder")
+            self.sigma = result.params["sigma"].value
+            self.beta = result.params["beta"].value
+            return self.sigma, self.beta
```

Comment:
¿Algo así, @islasfernando ?:
```suggestion
            return self.sigma, self.beta
        else:
            raise NameError('InvalidModel')
```

Reply:
@islasfernando : Otra opción sería quitar el `if` y nombrar esta función `def fit_detection_function_hazard(self)` y cuando en el futuro implementemos el segundo modelo (distinto a `hazard`) entonces agregamos `def fit_detection_function(self, model)` con el `if `. Para no adelantarnos al futuro.

Reply:
Va, lo voy a quitar entonces. Gracias @evaristor